src/main/kotlin/no/hvl/dat251/fjelltur/controller/RuleController.kt

```diff
@@ -42,4 +52,55 @@ class RuleController(@Autowired val ruleService: RuleService) {
   fun createDistance(@Valid @RequestBody request: CreateDistanceRuleRequest): DistanceRuleIdOnlyResponse {
     return ruleService.createDistanceRule(request).toDistanceRuleOnlyResponse()
   }
+
+  @Operation(
+    summary = "Delete a rule",
+    parameters = [Parameter(name = "name", description = "The name of the rule")],
+    responses = [
+      ApiResponse(
+        responseCode = "404",
+        description = "No rule with the given name in the database",
+        content = [Content()]
+      ),
+      ApiResponse(responseCode = "200")
+    ]
+  )
+  @DeleteMapping("/delete")
+  fun deleteRule(@RequestParam("name") name: String) {
+    return ruleService.deleteRule(name)
+  }
+
+  @Operation(
+    summary = "Update fields of a distance rule",
+    requestBody = SwaggerRequestBody(description = "The fields that is suppose to be updated, the fields that ase suppose to be the same can be null"),
+    responses = [
+      ApiResponse(responseCode = "200"),
+      ApiResponse(
+        responseCode = "404",
+        description = "No rule with the given name in the database",
+        content = [Content()]
+      )
+    ]
+  )
+  @PutMapping("/update/distance")
+  fun updateDistance(@Valid @RequestBody request: UpdateDistanceRuleRequest): DistanceRuleIdOnlyResponse {
+    return ruleService.updateDistanceRule(request).toDistanceRuleOnlyResponse()
+  }
+
+  @Operation(
+    summary = "Update fields of a time rule",
+    requestBody = SwaggerRequestBody(description = "The fields that is suppose to be updated, the fields that ase suppose to be the same can be null"),
+    responses = [
+      ApiResponse(responseCode = "200"),
+      ApiResponse(
+        responseCode = "404",
+        description = "No rule with the given name in the database",
+        content = [Content()]
+      )
+    ]
+  )
+  @PutMapping("/update/time")
+  fun updateTime(@Valid @RequestBody request: UpdateTimeRuleRequest): TimeRuleIdOnlyResponse {
+    return ruleService.updateTimeRule(request).toTimeRuleOnlyResponse()
+  }
```

Comment:
Legg til find by name controller

Reply:
@elgbar Har du en tricky måte å lage en fun som tar alle rules eller må hver type ha egen tenker du?

Reply:
Finn en rule fra db og type sjekk resultatet, dette kan være funksjon som kjører på getAll endpointet også 

``kotlin
with(val rule = ruleRepo.findByName(name)){
 is TimeRule -> rule.toResponse()
 is DistanceRule -> rule.toResponse()
else -> error("unknown rule")
}
```

Reply:
Her røyk mine kotlin kunnskaper...